packages/venia-ui/lib/components/Image/image.js

```diff
@@ -1,92 +1,162 @@
-import React, { Fragment, useMemo } from 'react';
-import { func, shape, string } from 'prop-types';
+import React from 'react';
+import PropTypes, {
+    bool,
+    func,
+    number,
+    oneOfType,
+    shape,
+    string
+} from 'prop-types';
 import { useImage } from '@magento/peregrine/lib/talons/Image/useImage';
 
-import { generateSrcset } from '../../util/images';
 import { mergeClasses } from '../../classify';
 import defaultClasses from './image.css';
+import PlaceholderImage from './placeholderImage';
+import ResourceImage from './resourceImage';
+import SimpleImage from './simpleImage';
```

Comment:
I know we still do this in a couple places (tiles come to mind), but when we're conditionally rendering child components, it's not ideal to synchronously import all of them up front. Not only are we loading more code than we need sooner than we need it, we're losing the ability to tree-shake these components.

Consider letting components choose the image variant they need without going through `Image`.

Reply:
Good points. I think I may create a separate issue for this, as this would be a non-trivial refactor at this point. The `Image` component takes care of all the placeholder handling, for example.

Reply:
Ok I created [JIRA 85](https://jira.corp.magento.com/browse/PWA-85) for this.